probes/aerospike/config.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	as "github.com/aerospike/aerospike-client-go"
+	"github.com/criteo/blackbox-prober/pkg/discovery"
+	"github.com/criteo/blackbox-prober/pkg/scheduler"
+)
+
+// Config used to configure the client of Aerospike
+type AerospikeClientConfig struct {
+	// auth
+	authEnabled  bool
+	authExternal bool
+	username     string
+	password     string
+	// tls
+	tlsEnabled    bool
+	tlsHostname   string
+	tlsSkipVerify bool
+	// Contact point
+	host as.Host
+}
+
+// Config used to configure the endpoint of Aerospike
+type AerospikeEndpointConfig struct {
+	AuthEnabled bool `yaml:"auth_enabled,omitempty"`
+	// If AuthEnabled use Aerospike auth external otherwise use internal
+	AuthExternal  bool `yaml:"auth_external,omitempty"`
+	TLSSkipVerify bool `yaml:"tls_skip_verify,omitempty"`
+	// ENV related config
+	// Env variable name to use to load credentials for Aerospike
+	UsernameEnv string `yaml:"username_env,omitempty"`
+	PasswordEnv string `yaml:"password_env,omitempty"`
+	// DISCOVERY related config
+	// Tag to use to determine if Aerospike need to be configured with TLS
+	TLSTag string `yaml:"tls_tag,omitempty"`
+	// Metadata key to get the Hostname to use for TLS auth (only used if tlsTag is set)
+	TLSHostnameMetaKey string `yaml:"tls_hostname_meta_key,omitempty"`
+}
+
+var (
+	defaultAerospikeClient = AerospikeEndpointConfig{
+		AuthEnabled:        true,
+		AuthExternal:       true,
+		UsernameEnv:        "AEROSPIKE_USERNAME",
```

Comment:
Why not getting both of them from config file?

Reply:
env variables used by a process can be read by doing cat /proc/pid/environ
so I'm not sure that pattern add any security

Reply:
/proc/ is in memory if I'm not mistaken.
Using env also prevent committing credentials by mistake. It is also simple to inject in both Mesos and Kubernetes (directly from a secret).

Reply:
/proc/ is in memory if I'm not mistaken -> yes but will be there any time the process run which mean always on k8s so I wanted to point that this doesn't provide any security